lightkurve/prf/prfmodel.py

```diff
@@ -108,6 +108,66 @@ def evaluate(self, center_col, center_row, flux=1., scale_col=1., scale_row=1.,
                                                  grid=False).reshape(self.shape)
         return self.prf_model
 
+    def gradient(self, center_col, center_row, flux=1., scale_col=1., scale_row=1.,
+                 rotation_angle=0.):
+        """
+        This function returns the gradient of the KeplerPRF model with
+        respect to center_col, center_row, flux, scale_col, scale_row,
+        and rotation_angle.
+
+        Parameters
+        ----------
+        center_col, center_row : float
+            Column and row coordinates of the center
+        flux : float
+            Total integrated flux of the PRF
+        scale_col, scale_row : float
+            Pixel scale in the column and row directions
+        rotation_angle : float
+            Rotation angle in radians
+
+        Returns
+        -------
+        grad_prf : list
+            Returns a list of arrays where the elements are the partial derivatives
+            of the KeplerPRF model with respect to center_col, center_row, flux, scale_col,
+            scale_row, and rotation_angle, respectively.
+        """
+        cosa = math.cos(rotation_angle)
+        sina = math.sin(rotation_angle)
+
+        delta_col = self.col_coord - center_col
+        delta_row = self.row_coord - center_row
+        delta_col, delta_row = np.meshgrid(delta_col, delta_row)
+
+        rot_row = delta_row * cosa - delta_col * sina
+        rot_col = delta_row * sina + delta_col * cosa
+
+        deriv_flux = self.interpolate(rot_row.flatten() * scale_row,
+                                      rot_col.flatten() * scale_col,
+                                      grid=False).reshape(self.shape)
+
+        interp_dy = self.interpolate(rot_row.flatten() * scale_row,
+                                     rot_col.flatten() * scale_col,
+                                     grid=False, dy=1).reshape(self.shape)
+
+        interp_dx = self.interpolate(rot_row.flatten() * scale_row,
+                                     rot_col.flatten() * scale_col,
+                                     grid=False, dx=1).reshape(self.shape)
+
+        scale_row_times_interp_dx = scale_row * interp_dx
+        scale_col_times_interp_dy = scale_col * interp_dy
+
+        deriv_center_col = - flux * (cosa * scale_col_times_interp_dy - sina * scale_row_times_interp_dx)
+        deriv_center_row = - flux * (sina * scale_col_times_interp_dy + cosa * scale_row_times_interp_dx)
+        deriv_scale_row = flux * interp_dx * rot_row
+        deriv_scale_col = flux * interp_dy * rot_col
+        deriv_rotation_angle = flux * (interp_dy * scale_col * (delta_row * cosa - delta_col * sina)
+                                       - interp_dx * scale_row * (delta_row * sina + delta_col * cosa))
```

Comment:
Can you add a few lines of comments to explain where these equations come from?

Reply:
I uploaded the proof [here](https://github.com/KeplerGO/lightkurve/files/2217633/gradients.pdf). I'm usually bad at notation, so let me know how cryptic it is.